### ISSUE TITLE
Restructure /frontend to better support reusable code shared by UI and OSSMC

### ISSUE THREAD

**Issue by jshaughn** [kiali/kiali#5916]
Epic https://github.com/kiali/openshift-servicemesh-plugin/issues/109

The OSSMC effort uses a large number of Kiali i-frames to present its Service Mesh integration.  For this effort to move from a preview to a GA level for OpenShift, the plugins must move from i-frames to native plugins.

The idea to support both native plugins and the standalone Kiali UI is to try and establish a set of core, shareable components.  Note that this is not planned to be a generally consumable set of generic components, as the only two consumers at this time are OSSMC plugins and Kiali standalone.  In fact, it may be found that even these two consumers may not be able to share as much as is hoped, time will tell whether it makes more sense to develop the two things separately, despite a lot of similarity.

This issue is for doing an initial restructuring of kiali/frontend to establish a new "core" area for the shared component code.

**Comment by aljesusg** [comment]
We discussed about this, it makes no sense pull all the code for only the components/types that we want to share it with other efforts and we should make so changes to do it and I don't know if this will be possible.

The options were create a monorepo o separate this in another repo as far as I remember.

I checked make this with lerna, npx etc...and the cleanest solution is this one. I don't know if we have other options we can discuss about it not problem.
